libraries-3/src/main/java/com/baeldung/barcodes/BarcodesController.java

```diff
@@ -0,0 +1,116 @@
+package com.baeldung.barcodes;
+
+import com.baeldung.barcodes.generators.BarbecueBarcodeGenerator;
+import com.baeldung.barcodes.generators.Barcode4jBarcodeGenerator;
+import com.baeldung.barcodes.generators.QRGenBarcodeGenerator;
+import com.baeldung.barcodes.generators.ZxingBarcodeGenerator;
+import com.google.zxing.WriterException;
+import net.sourceforge.barbecue.BarcodeException;
+import net.sourceforge.barbecue.output.OutputException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+
+@RestController
+@RequestMapping("/barcodes")
+public class BarcodesController {
+
+    //Barbecue library
+
+    @GetMapping(value = "/barbecue/upca", produces = MediaType.IMAGE_PNG_VALUE)
+    public ResponseEntity<BufferedImage> barbecueUPCABarcode(@RequestParam("barcodeText") String barcodeText) throws BarcodeException, OutputException {
```

Comment:
Can we not use `pathparam` rather than request param. This would be a cleaner REST interface, then.

Reply:
done